Engine/ac/route_finder.cpp

```diff
@@ -11,242 +11,91 @@
 // http://www.opensource.org/licenses/artistic-license-2.0.php
 //
 //=============================================================================
-//
-// New jump point search (JPS) A* pathfinder by Martin Sedlak.
-//
-//=============================================================================
 
 #include "ac/route_finder.h"
 
-#include <string.h>
-#include <math.h>
-
-#include "ac/common.h"   // quit()
-#include "ac/movelist.h"     // MoveList
-#include "ac/common_defines.h"
-#include "gfx/bitmap.h"
-
-#include "route_finder_jps.inl"
-
-extern MoveList *mls;
-
-#ifdef DEBUG_PATHFINDER
-extern void Display(const char *, ...);
-#endif
-
-using AGS::Common::Bitmap;
-namespace BitmapHelper = AGS::Common::BitmapHelper;
-
-#define MAKE_INTCOORD(x,y) (((unsigned short)x << 16) | ((unsigned short)y))
-
-static const int MAXNAVPOINTS = MAXNEEDSTAGES;
-static int navpoints[MAXNAVPOINTS];
-static int num_navpoints;
-static fixed move_speed_x, move_speed_y;
-static Navigation nav;
-
-void init_pathfinder()
+#include "ac/route_finder_impl.h"
+#include "ac/route_finder_impl_legacy.h"
+
+#include "debug/out.h"
+
+
+class IRouteFinder {
+public:
+virtual void init_pathfinder() = 0;
```

Comment:
I will use an opportunity to note again, that this is not the code style we are tend to use in the project. According to the code convention,
* the curve brackets are on their own line
* there should be indentation for members of class.
* member function names are in CamelCase.

Reply:
I'll tidy the style. It was an attempt to keep the boilerplate code small and tidy.

I would like to keep the member functions the same as the function names, to make it obvious during a text search that they are the same thing. The interface is hidden within the the cpp file, so it's not exposed to the rest of the project. If we ever do create a proper class to encapsulate the path finding then I would definitely prefer CamelCase for function names.

Reply:
> It was an attempt to keep the boilerplate code small and tidy.

The code style may be adjusted to this if you think it is beneficial, but having separate styles in chosen places is confusing.